### PR TITLE
Add design

### PR DESCRIPTION
**Pull request by juandspy**

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    participant Glitchtip-Jira integration
+    participant Jira
+
+    Team ->> app-interface: Defines an alert per Glitchtip project
+    Note over Team,app-interface: sets a webhook URL like<br/>example.com/api/v1/JIRA_PROJECT
+    app-interface ->> Glitchtip: Creates the alert
+
+    loop Every time alert triggers
+        Glitchtip->>Glitchtip-Jira integration: Issue info
+        Glitchtip-Jira integration->>Jira: Check if task exists?
```
Comment (chassing) [inline DESIGN.md:153]: Glitchtip calls the webhook for an alert [every minute](https://gitlab.com/glitchtip/glitchtip-backend/-/blob/master/glitchtip/settings.py?ref_type=heads#L449). I think we need to cache the `check if the task exists` respond for some time, would be worth to note somewhere. An in-memory cache should be suitable for that.

File: DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+```yaml
+alerts:
+- name: alert-name
+  ...
+  - provider: webhook  # send a Slack notification
+    url: https://glitchtip-jira-integration.example.com/api/v1/MY_JIRA_PROJECT
+```
+
+It's not even necessary to add one endpoint everytime a team creates a Jira project (which is not common tho), we can have a wildcard at `/api/v1/{jira_project}` and use the Jira project as a variable for interacting with the JIRA API.
```
Comment (chassing) [inline DESIGN.md]: `jira_project` must be the Jira project key

File: DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+The general webhook has this format:
+
+```js
+{
+    "alias": "Glitchtip",
+    "text": "GlitchTip Alert (${ISSUE_COUNT} issues)",
+    "attachments": [
+        {
+            "title": "${ISSUE_TITLE}",
+            "title_link": "${ISSUE_URL}",
```
Comment (chassing) [inline DESIGN.md:60]: It may be worth mentioning that we can use the `title_link` as a key to identify an issue. Maybe using it in a Jira label too.

**Review by chassing — APPROVED** [review]
Looks great to me. 🙇

**Comment by chassing** [comment]
If you don't mind, I'll use it for the AppSRE design-doc?

File: DESIGN.md (hunk 1 above)
Comment (juandspy) [reply DESIGN.md:153]: fixed in 72c5696128877a957143920100d86ceaaf631665

File: DESIGN.md (hunk 3 above)
Comment (juandspy) [reply DESIGN.md:60]: fixed in 5b6ca42c38ad5b09c24fd1eadfe872c9ff2e69a9

File: DESIGN.md (hunk 2 above)
Comment (juandspy) [reply DESIGN.md]: fixed in 8839be6ef8dd07287140897a6f66f64a2000f3d5

**Comment by juandspy** [comment]
> If you don't mind, I'll use it for the AppSRE design-doc?

sure! @chassing

**Pull request merged**